### PR TITLE
Fix/cmr 6856 catalog child links

### PR DESCRIPTION
**Pull request by matthewhanson**

### CODE REVIEW

File: search/lib/api/provider.js [hunk 1]
```diff
@@ -4,38 +4,42 @@ const { assertValid, schemas } = require('../validator');
...
+
+    const provider = {
+      id: providerId,
+      title: providerId,
+      description: `Root catalog for ${providerId}`,
+      stac_version: settings.stac.version,
+      links: [...links, ...childLinks]
+    };
     await assertValid(schemas.catalog, provider);
     response.status(200).json(provider);
```
Comment (jaybarra) [inline search/lib/api/provider.js:44]: `.status(200)` is redundant when using `.json` unless you need to set a different one

File: search/tests/api/provider.spec.js [hunk 2]
```diff
@@ -61,44 +61,46 @@ describe('getProviders', () => {
...
     const request = createRequest({
       params: {
-        providerId: 'LARC_ASDC'
+        providerId: 'USGS_EROS'
       }
     });
     const response = createMockResponse();
     await getProvider(request, response);
+    const dat = response.getData();
+    dat.json.links = dat.json.links.slice(0, 4);
```
Comment (jaybarra) [inline search/tests/api/provider.spec.js:103]: Is a test being done with this?

**Review by daniel-zamora — APPROVED** [review]

**Review by jaybarra — CHANGES_REQUESTED** [review]
I don't see the new test in the build results. If there are tests are deeper levels they might be be being executed.

**Review by jaybarra — APPROVED** [review]
Ignore the comment about the tests. Just remove the unused dat slice unless there is something testing it.

**Comment by matthewhanson** [comment]
> Ignore the comment about the tests. Just remove the unused dat slice unless there is something testing it.

@jaybarra This PR adds a bunch of 'child' links to every root provider catalog. Before, the provider catalog only contained a handful of deterministic links (self, root, parent, collections, search). But now there are links for all the collections in a provider.

I recognize that this makes it not a test for the child links, but I don't want to have a fixture of all the collections in a test provider because this could easily change. Maybe a test to check that there is some minimum # of child links would be sufficient.

**Pull request merged**